client/src/components/continuousLegend/index.js

```diff
@@ -115,7 +115,10 @@ class ContinuousLegend extends React.Component {
       prevProps.colorScale !== colorScale
     ) {
       /* always remove it, if it's not continuous we don't put it back. */
-      d3.select("#continuous_legend").selectAll("*").remove();
+      const allLegend = d3.select("#continuous_legend").selectAll("*");
```

Comment:
Suggest using optional chaining for clarity, ie,  `d3.select("#continuous_legend")?.selectAll("*")?.remove();`